app/pods/components/agenda/agenda-header/component.js

```diff
@@ -240,71 +247,84 @@ export default Component.extend(FileSaverMixin, {
     },
 
     async approveAndCloseAgenda(session) {
-      const isApprovable = await this.currentAgenda.get('isApprovable');
-      const meetingOfAgenda = await this.currentAgenda.get('createdFor');
-      const agendasOfMeeting = await meetingOfAgenda.get('agendas');
-      if (!isApprovable) {
-        this.set('showWarning', true);
-      } else {
-        if (this.get('isApprovingAgenda')) {
-          return;
-        }
-        this.set('isApprovingAgenda', true);
-        this.changeLoading();
-        const agendaToLock = await agendasOfMeeting.find((agenda) => agenda.get('isDesignAgenda'));
-
-        agendaToLock.set(
-          'modified',
-          moment()
-            .utc()
-            .toDate()
-        );
-        agendaToLock.save().then(async(agendaToApprove) => {
-          await this.agendaService.approveAgenda(session, agendaToApprove);
-          return agendaToApprove;
-        })
-          .then(async(agendaToApprove) => {
-            // We reloaden de agenda hier om de recente changes m.b.t. het approven van de agenda binnen te halen
-            const reloadedAgenda = await this.store.findRecord('agenda', agendaToApprove.get('id'), {
-              reload: true,
-              include: 'status',
-            });
-            await this.lockAgenda(reloadedAgenda);
+      const canCloseAgenda = await this.canCloseAgenda();
+      if (canCloseAgenda) {
+        const isApprovable = await this.currentAgenda.get('isApprovable');
+        const meetingOfAgenda = await this.currentAgenda.get('createdFor');
+        const agendasOfMeeting = await meetingOfAgenda.get('agendas');
+        if (!isApprovable) {
+          this.set('showWarning', true);
+        } else {
```

Comment:
Is het niet deze flag dat die side-effects creert ?
de isApprovable zorgt hier voor de popup met de showWarning, en die popup is gelinkt aan doApproveAgenda

Reply:
die hele canCloseAgenda is volgens mij een copy van die Approvable nu.

In die method approveAndCloseAgenda wil je volgens mij gewoon afbreken als de isApprovable false is en de pop tonen dat de agenda niet kan worden afgesloten omdat er nog punten 'nog niet formeel ok' zijn.

Reply:
fixed